java-checks/src/main/java/org/sonar/java/checks/ArrayHashCodeAndToStringCheck.java

```diff
@@ -40,7 +52,41 @@ protected MethodMatchers getMethodInvocationMatchers() {
 
   @Override
   protected void onMethodInvocationFound(MethodInvocationTree mit) {
-    String methodName = mit.methodSymbol().name();
-    reportIssue(ExpressionUtils.methodName(mit), "Use \"Arrays." + methodName + "(array)\" instead.");
+    IdentifierTree methodTree = ExpressionUtils.methodName(mit);
+    String methodName = methodTree.name();
+    String methodCallee = QuickFixHelper.contentForTree(((MemberSelectExpressionTreeImpl) mit.methodSelect()).expression(), context);
+    QuickFixHelper.newIssue(context)
+      .forRule(this)
+      .onTree(methodTree)
+      .withMessage("Use \"Arrays." + methodName + "(array)\" instead.")
+      .withQuickFix(() -> getQuickFix(mit, methodName, methodCallee))
+      .report();
   }
+
+  private JavaQuickFix getQuickFix(MethodInvocationTree tree, String methodName, String methodCallee) {
+    List<JavaTextEdit> edits = new ArrayList<>();
+    getImportSupplier()
+      .newImportEdit(ARRAYS)
+      .ifPresent(edits::add);
+
+    if ("toString".equals(methodName)) {
+      edits.add(JavaTextEdit.replaceTree(tree, "Arrays.toString(" + methodCallee + ")"));
+      return JavaQuickFix.newQuickFix("Use \"Arrays.toString(array)\" instead.")
+        .addTextEdits(edits)
+        .build();
+    } else {
+      edits.add(JavaTextEdit.replaceTree(tree, "Arrays.hashCode(" + methodCallee + ")"));
+      return JavaQuickFix.newQuickFix("Use \"Arrays.hashCode(array)\" instead.")
+        .addTextEdits(edits)
+        .build();
+    }
+  }
+
+  private QuickFixHelper.ImportSupplier getImportSupplier() {
+    if (importSupplier == null) {
```

Comment:
I believe that we have a problem here.

Since you create an `ImportSupplier` based on the context, it means that there is a direct link between the current file and the rule. When changing to another file, the context will change but the `ImportSupplier` will stay the same. I believe this might (will?) lead to errors when generating the quickfix.

If you look at other rules using an import supplier, they reset to null the supplier after each file.